### PR TITLE
chore: attempt to create a bundle analysis file

### PR DESCRIPTION
**Pull request by ovflowd**
This PR attempts to add bundle analysis within the Pull Request GitHub Action. This uses https://github.com/hashicorp/nextjs-bundle-analysis

### CODE REVIEW

**Comment by bmuenzenmeyer** [comment]
@ovflowd this is cool! Should we combine this with the real user metrics capabilities of [Lighthouse CI](https://github.com/GoogleChrome/lighthouse-ci)? That would give us a nice automated window into measurable UX and regressions for relatively little cost (slower CI)

File: .github/workflows/pull-request.yml
Comment (bmuenzenmeyer) [inline .github/workflows/pull-request.yml:44]: If we want to streamline these flags, I *think* the GH Actions will respect entries in a local `.npmrc` file

```
audit=false
fund=false
```

**Review by bmuenzenmeyer — APPROVED** [review]
small future idea inside

File: .github/workflows/pull-request.yml
Comment (ovflowd) [reply .github/workflows/pull-request.yml:44]: I would rather keep those `ci` only. I do think that for every environment besides CI fund flags and audit reports are important.

**Comment by ovflowd** [comment]
> @ovflowd this is cool! Should we combine this with the real user metrics capabilities of [Lighthouse CI](https://github.com/GoogleChrome/lighthouse-ci)? That would give us a nice automated window into measurable UX and regressions for relatively little cost (slower CI)

Sure, and that sounds awesome, but I am still determining how efficient the solution of this PR is. We would need to see an example of this running. Also, it only supports Next.js.

I'm worried about this not working or the analysis not being good enough. I've found an example of this being used: https://github.com/MaciejGarncarski/photo-app/pull/38#issuecomment-1407463036 another example, https://github.com/hashicorp/dev-portal/pull/2050#issuecomment-1623639203

**Comment by bmuenzenmeyer** [comment]
I imagine this solution doesn't work until you have a baseline.

well, lighthouse will test the site as a user would experience it using standard web-vitals metrics. Those metrics are perhaps more valuable. Really, they ALL are valuable, as the value comes from an awareness as developers on the impact of our actions... and if we are okay with the delta's they produce.

**Comment by ovflowd** [comment]
(I wrote this before your last comment, Brian)

Also, since our pages are dynamically generated, the Markdown content of each page is not taken into consideration on the page-size. So the extra Markdown (MDX) content might add a few KB's depending of the page.

But since those are part of the content-itself, they're not part of the "bundle", as on the server-side rendering they're rendered as HTML and not part of the bundle itself.

Worth mentioning that, these static props are probably loaded within the page on pre-fetch or when the page is invoked directly (Example: https://vercel.nodejs.org/_next/data/7UowjOarLsTee3Vh3lvww/en.json?path=en)

But I wouldn't take these into consideration, to be honest.

**Comment by ovflowd** [comment]
> well, lighthouse will test the site as a user would experience it using standard web-vitals metrics. Those metrics are perhaps more valuable. Really, they ALL are valuable, as the value comes from an awareness as developers on the impact of our actions... and if we are okay with the delta's they produce.

Yeah, that's what matters.

> I imagine this solution doesn't work until you have a baseline.

Seeing some of the examples (usages of this Action out in the wild), it seems worthwhile. But it's just part of the equation.

**Comment by bmuenzenmeyer** [comment]
also worth mentioning, if we are looking for streamlined tools that don't ADD more to our chain, that https://vercel.com/analytics exists but I've never used it nor know if it works with pull requests.

**Comment by ovflowd** [comment]
> also worth mentioning, if we are looking for streamlined tools that don't ADD more to our chain, that [vercel.com/analytics](https://vercel.com/analytics) exists but I've never used it nor know if it works with pull requests.

Yes, those aren't for PRs, sadly.

**Comment by bmuenzenmeyer** [comment]
I'll defer to you if you think this is valuable. I intend to introduce lighthouse-ci when I have my next chunk of bigger free time

**Comment by ovflowd** [comment]
Let's merge this; if we don't see value here, we can always revert this.

We can test this by changing a Component that is used within the current Website and see if it generates a report.

**Comment by ovflowd** [comment]
I can also just run it locally against an older version of the website and paste the results here 🤔

**Comment by ovflowd** [comment]
## 📦 Next.js Bundle Analysis for nodejs.org

This analysis was generated by the [Next.js Bundle Analysis action](https://github.com/hashicorp/nextjs-bundle-analysis). 🤖

### ⚠️  Global Bundle Size Increased
  
Page | Size (compressed) | 
|---|---|
| `global` | `136.46 KB` _(🟢 901 B)_ |

<details>
<summary>Details</summary>
<p>The <strong>global bundle</strong> is the javascript bundle that loads alongside every page. It is in its own category because its impact is much higher - an increase to its size means that every page on your website loads slower, and a decrease means every page loads faster.</p>
<p>Any third party scripts you have added directly to your app using the <code>&lt;script&gt;</code> tag are not accounted for in this analysis</p>
<p>If you want further insight into what is behind the changes, give <a href='https://www.npmjs.com/package/@next/bundle-analyzer'>@next/bundle-analyzer</a> a try!</p>
</details>

### Two Pages Changed Size
  
The following pages changed size from the code in this PR compared to its base branch:

Page | Size (compressed) | First Load |
|---|---|---|
| `/404` | `34.1 KB` _(🟢 -2.87 KB)_ | 170.56 KB |
| `/[...path]` | `34.05 KB` _(🟢 -2.86 KB)_ | 170.51 KB |

<details>
<summary>Details</summary>
<p>Only the gzipped size is provided here based on <a href='https://twitter.com/slightlylate/status/1412851269211811845'>an expert tip</a>.</p>
<p><strong>First Load</strong> is the size of the global bundle plus the bundle for the individual page. If a user were to show up to your website and land on a given page, the first load size represents the amount of javascript that user would need to download. If <code>next/link</code> is used, subsequent page loads would only need to download that page's bundle (the number in the "Size" column), since the global bundle has already been downloaded.</p>
<p>Any third party scripts you have added directly to your app using the <code>&lt;script&gt;</code> tag are not accounted for in this analysis</p>
<p>Next to the size is how much the size has increased or decreased compared with the base branch of this PR. If this percentage has increased by undefined% or more, there will be a red status indicator applied, indicating that special attention should be given to this.
</details>

**Comment by ovflowd** [comment]
☝️ comment above is me manually running the stuff locally and pasting the output on the comment above.

**Comment by ovflowd** [comment]
cc @nodejs/website requesting fast-tracking here so that we can test these changes on currently opened PRs.

**Comment by bmuenzenmeyer** [comment]
> cc @nodejs/website requesting fast-tracking here so that we can test these changes on currently opened PRs.

✅

**Comment by ovflowd** [comment]
> > cc @nodejs/website requesting fast-tracking here so that we can test these changes on currently opened PRs.
> 
> ✅

A thumbs up on the comment usually is enough 👀

**Pull request merged**
